tests/tracing/test_splining.py

```diff
@@ -0,0 +1,163 @@
+# Disable ruff 301 - pickle loading is unsafe, but we don't care for tests
+# ruff: noqa: S301
+"""Test the splining module."""
+
+import pickle
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from topostats.tracing.splining import splining_image
+
+BASE_DIR = Path.cwd()
+RESOURCES = BASE_DIR / "tests" / "resources"
+
+# pylint: disable=unspecified-encoding
+# pylint: disable=too-many-locals
+# pylint: disable=too-many-arguments
+
+
+@pytest.mark.parametrize(
+    (
+        "image_filename",
+        "ordered_tracing_direction_data_filename",
+        "pixel_to_nm_scaling",
+        "filename",
+        "expected_all_splines_data_filename",
+        "expected_grainstats_additions_filename",
+        "expected_molstats_filename",
+    ),
+    [
+        pytest.param(
+            "example_catenanes.npy",
+            "example_catenanes_ordered_tracing_data.pkl",
+            1.0,  # pixel_to_nm_scaling
+            "catenane",  # filename
+            "example_catenanes_splining_data.pkl",
+            "example_catenanes_splining_grainstats_additions.csv",
+            "example_catenanes_splining_molstats.csv",
+            id="catenane",
+        ),
+        pytest.param(
+            "example_rep_int.npy",
+            "example_rep_int_ordered_tracing_data.pkl",
+            1.0,  # pixel_to_nm_scaling
+            "replication_intermediate",  # filename
+            "example_rep_int_splining_data.pkl",
+            "example_rep_int_splining_grainstats_additions.csv",
+            "example_rep_int_splining_molstats.csv",
+            id="replication_intermediate",
+        ),
+    ],
+)
+def test_splining_image(
+    image_filename: str,
+    ordered_tracing_direction_data_filename: str,
+    pixel_to_nm_scaling: float,
+    filename: str,
+    expected_all_splines_data_filename: str,
+    expected_grainstats_additions_filename: str,
+    expected_molstats_filename: str,
+) -> None:
+    """Test the splining_image function of the splining module."""
+    # Load the data
+    image = np.load(RESOURCES / image_filename)
+
+    # Load the ordered tracing direction data
+    with Path.open(RESOURCES / ordered_tracing_direction_data_filename, "rb") as file:
+        ordered_tracing_direction_data = pickle.load(file)
+
+    result_all_splines_data, result_grainstats_additions_df, result_molstats_df = splining_image(
+        image=image,
+        ordered_tracing_direction_data=ordered_tracing_direction_data,
+        pixel_to_nm_scaling=pixel_to_nm_scaling,
+        filename=filename,
+        method="rolling_window",
+        rolling_window_size=20e-9,
+        spline_step_size=7.0e-9,
+        spline_linear_smoothing=5.0,
+        spline_circular_smoothing=5.0,
+        spline_degree=3,
+    )
+
+    # # Debugging
```

Comment:
I'd be inclined to keep it as-is if possible because:

1 - I really think it's useful since we are going to be updating this test a lot over the next few weeks probably. I'd only end up re-writing it each time I update the test, slowing down test updates.

2 - I can't think of anywhere else in our codebase that we would need to debug viewing the splines like this and it would take some time to make a function out of it and test it.

That being said it is a lot of code mess and I very much empathise with wanting to get rid of commented-out code since it does tend to linger infinitely

Reply:
Perhaps a middle ground of making it a function within the test?

It would then be possible to have a single line which calls the function that can be commented/uncomment the call to the function for debugging and the test file doesn't have a large amount of lingering code.

I'd did similar with a plotting function I'd written for the `topostats.measure.feret.plot_feret()`](https://github.com/AFM-SPM/TopoStats/blob/70d02bd811285efcb6c72bdef854b148f414b951/topostats/measure/feret.py#L430) which I used for debugging. I already had images that I'd looked at during the debugging process so [tests](https://github.com/AFM-SPM/TopoStats/blob/70d02bd811285efcb6c72bdef854b148f414b951/tests/measure/test_feret.py#L1308) were relatively straight-forward to add.

Reply:
Done 👍 good suggestion. That does clean up the code nicely.